src/DataStructures/Tensor/Expressions/Product.hpp

```diff
@@ -271,3 +271,25 @@ SPECTRE_ALWAYS_INLINE auto operator*(
   return TensorExpressions::NumberAsExpression(number) * t;
 }
 // @}
+
+/// \ingroup TensorExpressionsGroup
+/// \brief Returns the tensor expression representing the quotient of a tensor
+/// expression and a `double`
```

Comment:
Could you add a `\note` that the implementation switches to `* (1.0 / number)`? While that won't matter most of the time, it may matter to someone, so they should be able to find that out :)